src/lib/mina_base/parties_logic.ml

```diff
@@ -33,6 +33,8 @@ module type Bool_intf = sig
 
   val all : t list -> t
 
+  type single_failure_status
```

Comment:
could we make this type `failure_status`, and the other one `failure_status_tbl`?